ListApp/Sources/ListApp.Contracts/Interfaces/IUpdateItemService.cs

```diff
@@ -5,7 +5,7 @@ namespace ListApp.Contracts.Interfaces
 {
     public interface IUpdateItemService
     {
-        Task<OperationResult> UpdateItemAsync(ListItem oldItem, ListItem newItem);
-        Task<OperationResult> CheckIfItemExistsAsync(ListItem item);
+        Task<ListItemDbOperationResult> UpdateItemAsync(ListItem newItem);
+        Task<ListItemDbOperationResult> PrepareUpdatedItemAsync(ListItem newItem);
```

Comment:
I'm afraid this is not a very robust API design. Nothing forces the caller to call `PrepareUpdatedItemAsync` before s/he calls `UpdateItemAsync`. He wants to update and finds an update in the Intellisense - he may not be aware of some method he should call first. Here it doesn't even seem to be such a big deal. Due to the implementation, however, such call would bypass the whole timestamp generation. Ideally, your API should not allow its users (other programmers) to do things they shouldn't do. And just in case this came up in your mind - XML documentation is the last resort workaround, not a solution :)